x/fungible/keeper/gas_coin_and_pool.go

```diff
@@ -18,16 +18,28 @@ import (
 // SetupChainGasCoinAndPool setup gas ZRC20, and ZETA/gas pool for a chain
 // add 0.1gas/0.1wzeta to the pool
 // FIXME: add cointype and use proper gas limit based on cointype/chain
-func (k Keeper) SetupChainGasCoinAndPool(ctx sdk.Context, chainID int64, gasAssetName string, symbol string, decimals uint8) (ethcommon.Address, error) {
+func (k Keeper) SetupChainGasCoinAndPool(
+	ctx sdk.Context,
+	chainID int64,
+	gasAssetName string,
+	symbol string,
+	decimals uint8,
+	gasLimit *big.Int,
+) (ethcommon.Address, error) {
 	chain := common.GetChainFromChainID(chainID)
 	if chain == nil {
 		return ethcommon.Address{}, zetaObserverTypes.ErrSupportedChains
 	}
 	name := fmt.Sprintf("%s-%s", gasAssetName, chain.ChainName)
 
-	transferGasLimit := big.NewInt(21_000)
-	if common.IsBitcoinChain(chain.ChainId) {
-		transferGasLimit = big.NewInt(100) // 100B for a typical tx
+	transferGasLimit := gasLimit
+
+	// default values
+	if transferGasLimit == nil {
+		transferGasLimit = big.NewInt(21_000)
+		if common.IsBitcoinChain(chain.ChainId) {
+			transferGasLimit = big.NewInt(100) // 100B for a typical tx
```

Comment:
maybe we can use a median/average gas limit 1600B?

Reply:
It's what we use right now for smoke test. For Bitcoin mainnet, it will use the gas limit set in the `DeployZRC20` message